src/CodeGen_Hexagon.cpp

```diff
@@ -549,14 +549,24 @@ Value *CodeGen_Hexagon::slice_vector(Value *vec, int start, int size) {
     // If we're getting a native vector bits worth of data from half
     // of the argument, we might be able to use lo/hi if the start is appropriate.
     if (size*2 == vec_elements && element_bits*size == native_vector_bits()) {
-        CallInst *call = dyn_cast<CallInst>(vec);
+        CallInst *call;
+        if (BitCastInst *bc = dyn_cast<BitCastInst>(vec)) {
+            call = dyn_cast<CallInst>(bc->getOperand(0));
+        } else {
+            call = dyn_cast<CallInst>(vec);
+        }
         llvm::Function *combine_fn =
             Intrinsic::getDeclaration(module.get(), IPICK(Intrinsic::hexagon_V6_vcombine));
-        if (call && call->getCalledFunction() == combine_fn) {
+        llvm::Type *ret_ty = llvm::VectorType::get(vec_ty->getScalarType(), size);
+        bool is_vcombine = call && call->getCalledFunction() == combine_fn;
+        if (is_vcombine) {
             if (start == 0) {
-                return call->getArgOperand(1);
+                // Not bitcasting here puts the onus on the caller to use call_intrin_cast before
```

Comment:
I'd still like to remove the comment about call_intrin_cast. The reason I commented previously is that the comment made me think that this function is changing the type, when it's not. The reason that matter is because this function could be called by CodeGen_LLVM, where it really expects precisely the behavior that the later part of the comment describes. The comment suggests that some of that behavior might break. In fact, I'm not sure a comment here at all is useful, the only reason there should be a comment is if it did something surprising, like *not* returning a vector of the type of the input.